debezium-core/src/main/java/io/debezium/config/ValidationResults.java

```diff
@@ -14,11 +14,11 @@
 
 public class ValidationResults {
 
-    public final List<ValidationResult> validationResults;
     public Status status;
+    public final List<ValidationResult> validationResults;
 
-    public ValidationResults(Connector connector, Map<String, ?> properties) {
-        this.validationResults = convertConfigToValidationResults(connector.validate(convertPropertiesToStrings(properties)));
+    public ValidationResults(ConnectorParameter connector, Map<String, ?> properties) {
+        this.validationResults = convertConfigToValidationResults(connector.get().validate(convertPropertiesToStrings(properties)));
```

Comment:
Although this was already in... I think that the result of `connector.get().validate()` should be a parameter for this class' constructor or something. I would honestly not expect a class named `ValidationResult` to internally call the validation, even less that it would be done in constructor.

Reply:
What else should we call it?

Validation must happen inside this class as we might have to potentially catch exceptions here too and this is shared behavior.

Reply:
I just just find it unexpected that the validation is performed in an instance of something called `ValidationResult`.... would kinda expect this instance to be create only after the validation happened.

Reply:
I will rename those classes to `ConnectorConfigValidator`  and `FilterConfigValidator`. I think that matches better the current usage. Initially those classes were only used as a data model for the results. That's the reason for those names. But I agree, they are outdated for the different approach that encapsulates repetitive code that is used for the validation. wdyt?

Reply:
Done